jlibra-core/src/main/java/dev/jlibra/serialization/Serializer.java

```diff
@@ -0,0 +1,75 @@
+package dev.jlibra.serialization;
+
+import static java.nio.ByteOrder.LITTLE_ENDIAN;
+
+import java.nio.ByteBuffer;
+import java.nio.charset.Charset;
+import java.util.List;
+
+import dev.jlibra.admissioncontrol.transaction.TransactionArgument;
+
+public class Serializer {
+
+    private byte[] bytes;
+
+    private Serializer(byte[] bytes) {
+        this.bytes = bytes;
```

Comment:
See my comment below for longer description but I think this should copy input `bytes` into new `bytes[]` and don't work on the object passed in to get some thread safety.

Reply:
But this can only be called by the class itself as it is `private` (and is only called in the builder), so I don't see a problem here. If I get this right, the actually executing stuff is in `append` and this always copies the byte array.

Reply:
I see your point here. I didn't notice it was `private`. In this case all lines:
```
    private byte[] bytes;

    private Serializer(byte[] bytes) {
        this.bytes = bytes;
    }

    public static Serializer builder() {
        return new Serializer(new byte[0]);
    }
```
could be replaced with 
```
private final byte[] bytes = new byte[0];
```
unless we really need that builder() method.